src/audio/iir.c

```diff
@@ -89,44 +94,45 @@ int32_t iir_df2t(struct iir_state_df2t *iir, int32_t x)
 			 * Q2.30 x Q1.31 -> Q3.61
 			 * Shift Q3.61 to Q3.31 with rounding
 			 */
-			acc = ((int64_t) iir->coef[c + 4]) * in + iir->delay[d];
-			tmp = (int32_t) Q_SHIFT_RND(acc, 61, 31);
+			acc = ((int64_t)iir->coef[c + 4]) * in + iir->delay[d];
+			tmp = (int32_t)Q_SHIFT_RND(acc, 61, 31);
 
 			/* Compute 1st delay */
 			acc = iir->delay[d + 1];
-			acc += ((int64_t) iir->coef[c + 3]) * in; /* Coef  b1 */
-			acc += ((int64_t) iir->coef[c + 1]) * tmp; /* Coef a1 */
+			acc += ((int64_t)iir->coef[c + 3]) * in; /* Coef  b1 */
+			acc += ((int64_t)iir->coef[c + 1]) * tmp; /* Coef a1 */
 			iir->delay[d] = acc;
 
 			/* Compute 2nd delay */
-			acc = ((int64_t) iir->coef[c + 2]) * in; /* Coef  b2 */
-			acc += ((int64_t) iir->coef[c]) * tmp; /* Coef a2 */
+			acc = ((int64_t)iir->coef[c + 2]) * in; /* Coef  b2 */
+			acc += ((int64_t)iir->coef[c]) * tmp; /* Coef a2 */
 			iir->delay[d + 1] = acc;
 
 			/* Gain, output shift, prepare for next biquad
 			 * Q2.14 x Q1.31 -> Q3.45, shift too Q3.31 and saturate
 			 */
-			acc = ((int64_t) iir->coef[c + 6]) * tmp; /* Gain */
+			acc = ((int64_t)iir->coef[c + 6]) * tmp; /* Gain */
 			acc = Q_SHIFT_RND(acc, 45 + iir->coef[c + 5], 31);
 			in = sat_int32(acc);
 			c += 7; /* Next coefficients section */
 			d += 2; /* Next biquad delays */
 		}
 		/* Output of previous section is in variable in */
-		out = sat_int32((int64_t) out + in);
+		out = sat_int32((int64_t)out + in);
 	}
 	return out;
 }
 
-size_t iir_init_coef_df2t(struct iir_state_df2t *iir, int32_t config[])
+size_t iir_init_coef_df2t(struct iir_state_df2t *iir,
+			  struct sof_eq_iir_header_df2t *config)
 {
-	iir->mute = 0;
-	iir->biquads = (int) config[0];
-	iir->biquads_in_series = (int) config[1];
-	iir->coef = &config[0]; /* TODO: Could change this to config[2] */
+	iir->biquads = config->num_sections;
+	iir->biquads_in_series = config->num_sections_in_series;
+	iir->coef = config->biquads;
 	iir->delay = NULL;
 
-	if ((iir->biquads > IIR_DF2T_BIQUADS_MAX) || (iir->biquads < 1)) {
+	if (iir->biquads > SOF_EQ_IIR_DF2T_BIQUADS_MAX ||
+	    iir->biquads < 1) {
```

Comment:
==0

Reply:
It's signed int and my old habit of always avoiding unsigned types so I did it this way. There's some multiplications done with the count but since it's not done in algo core (overhead from mixing signed and unsigned to arithmetic) it's no issue to change it to unsigned and compare to 0.